src/components/shared/SearchFilm.js

```diff
@@ -0,0 +1,69 @@
+import React, { Component, Fragment } from 'react'
+import { InputGroup, FormControl, Button } from 'react-bootstrap';
+import { connect } from 'react-redux';
+import { fetchMovies } from '../../redux/actions/moviesActions';
+import ButtonsCriteriaSearch from './ButtonsCriteriaSearch';
+import styled from 'styled-components'
+
+const StyledGroup = styled(InputGroup)`
+    padding: 0 50px;
+    margin: 0 auto;
+    max-width: 700px;
+
+    .input-group-append {
+        background-color: bisque;
+    }
+`;
+
+class SearchFilm extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            kind: 'Search',
+            left: 'Title',
+            right: 'Genre',
+            disabled: true
+        }
+    }
+
+    handleClick() {
+        this.props.dispatch(
+            fetchMovies(this.props.sort, this.props.search, this.myInput.value));
+    }
+
+    handleChange() {
+        const value = this.myInput.value;
+
+        value ? this.setState({disabled: false})
+              : this.setState({disabled: true})
+    }
+
+    render() {
+        return (
+            <Fragment>
+                <Button className="add-movie" variant="outline-danger" onClick={this.handleClick.bind(this)} disabled={this.state.disabled}>+ Add movie</Button>
```

Comment:
All the handlers should be bound in the class constructor.
React docs: 
"We generally recommend binding in the constructor or using the class fields syntax, to avoid this sort of performance problem."